### PR TITLE
[FIXED JENKINS-30926] fieldName could be blank

### PR DESCRIPTION
**Pull request by lordofthejars**
Resolves issue [JENKINS-30926](https://issues.jenkins-ci.org/browse/JENKINS-30926) by adding a message in case file name is blank or null.

@reviewbybees

### CODE REVIEW

**Comment by recena** [comment]
@lordofthejars It would be great to add a test.

**Comment by lordofthejars** [comment]
@recena Yes but well considering that the whole plugin is not tested directly, I am not sure that writing a test that checks if one field is null an exception is thrown improves so much the coverage. But yes adding tests not only for this fix would be great.

**Comment by recena** [comment]
@lordofthejars There is always a first test :dizzy_face:

**Comment by varmenise** [comment]
:bee: for the change itself. :ant: for the lack of test,  it s a fairly easy scenario to test

**Comment by lordofthejars** [comment]
@varmenise I know but it is a null check and the project itself has no test, I can add a test that checks null but I would say that the quality in terms of test would be the same.

**Comment by varmenise** [comment]
@lordofthejars yes that s why I gave you a bee :)

File: pom.xml [hunk 1]
```diff
@@ -42,6 +42,11 @@
             <artifactId>credentials</artifactId>
             <version>1.21</version>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (jtnord) [inline pom.xml]: <scope>test<scope>

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
+package org.jenkinsci.plugins.plaincredentials;
```
Comment (jtnord) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: missing (c)

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.UnsupportedEncodingException;
+
+public class FileCredentialsTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowAnExceptionIfFileNameIsBlank() throws IOException {
+        FileCredentials fileCredentials = new FileCredentialsImpl(CredentialsScope.GLOBAL, "1", "", new StubFileItem(), "", "");
+    }
+
+    private class StubFileItem implements FileItem {
```
Comment (jtnord) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: wouldthis not have been easier to just use [DiskFileItem](https://commons.apache.org/proper/commons-fileupload/apidocs/org/apache/commons/fileupload/disk/DiskFileItem.html#DiskFileItem(java.lang.String, java.lang.String, boolean, java.lang.String, int, java.io.File)) with the appropriate args?

Comment (lordofthejars) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: Yes it is a possibility but I used this approach to avoid having to set all the parameters in the constructor.

**Comment by jtnord** [comment]
:bee:

**Comment by lordofthejars** [comment]
@reviewbybees done

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java (hunk 2 above)
Comment (recena) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: Copyright is wrong.

**Comment by recena** [comment]
:ant: for lack of `@Issue` and wrong copyright

File: pom.xml [hunk 4]
```diff
@@ -42,6 +42,12 @@
             <artifactId>credentials</artifactId>
             <version>1.21</version>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
Comment (jglick) [inline pom.xml]: :ant: You do not need to specify a `junit` test dependency, it is automatic for all plugins.

File: src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java [hunk 5]
```diff
@@ -67,6 +67,12 @@
             this.fileName = fileName;
             this.data = Base64.decodeBase64(data);
         }
+        if (this.fileName == null || this.fileName.isEmpty()) {
+            throw new IllegalArgumentException(
+                    String.format("No FileName was provided or resolved. " +
+                            "Input file item was %s and input file name was %s.", file.toString(), fileName)
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java]: Printing `fileName` in the error message is not very useful here since we just determined it was blank.

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+
+public class FileCredentialsTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    @Issue("JENKINS-30926")
+    public void shouldThrowAnExceptionIfFileNameIsBlank() throws IOException {
+        FileCredentials fileCredentials = new FileCredentialsImpl(CredentialsScope.GLOBAL, "1", "",
+                new StubFileItem(), "", "");
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: BTW the test is not terribly helpful here because we are not really sure where the blank name came from to begin with. A browser bug? A problem in `config.jelly`?

**Comment by jglick** [comment]
:bee:

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java (hunk 6 above)
Comment (lordofthejars) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: I know but all upper comments were about having this test.

File: src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java (hunk 5 above)
Comment (recena) [inline src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java]: +1

Comment (lordofthejars) [inline src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java]: Ii can be different, I mean you can go there because `this.fileName = name.replaceFirst("^.+[/\\\\]", "");`gets blank but not because the entry fileName (the method attribute) which can be blank or not.

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java (hunk 6 above)
Comment (recena) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: Not exactly. I proposed add a test, not its implementation. Anyway, I prefer this test than not have it at all

File: src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java (hunk 5 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/plaincredentials/impl/FileCredentialsImpl.java]: Ah, never mind, I did not notice that `fileName` and `this.fileName` are potentially different at this point.

File: src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java (hunk 6 above)
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java]: Sure, not complaining about the existence of the test, just noting that it is not capturing what really went wrong here.

**Comment by lordofthejars** [comment]
@reviewbybees done
